httpcore5/src/main/java/org/apache/hc/core5/util/TimeValue.java

```diff
@@ -371,13 +347,43 @@ public int toSecondsIntBound() {
         return asBoundInt(toSeconds());
     }
 
+    public Timeout toTimeout() {
+        return Timeout.of(duration, timeUnit);
+    }
+
     @Override
-    public String toString() {
-        return String.format(Locale.ROOT, "%,d %s", duration, timeUnit);
+    public int compareTo(final TimeValue other) {
+        if (other != null) {
+            final TimeUnit targetTimeUnit = min(other.getTimeUnit());
+            return Long.compare(convert(targetTimeUnit), other.convert(targetTimeUnit));
+        } else {
+            return -Integer.MAX_VALUE;
+        }
     }
 
-    public Timeout toTimeout() {
-        return Timeout.of(duration, timeUnit);
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof TimeValue) {
+            final TimeValue that = (TimeValue) obj;
+            return this.duration == that.duration && LangUtils.equals(this.timeUnit, that.timeUnit);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Why not `Objects.hash(...)`?

Reply:
@michael-o For the sake of consistency with the rest of code base, though I am open to the idea of dropping our `LangUtils` in favor to `Objects`.

Reply:
Alright, should be taken in a separate PR.